master/getting-started/kubernetes/tutorials/advanced-policy.md

```diff
@@ -151,107 +111,148 @@ spec:
+<title>Welcome to nginx!</title>...
 ```
 
-After creating the policy, we can now access the nginx Service.  We also see that the pod can
-access google.com on the public internet.  This is because we have not defined any egress policy.
+After creating the policy, we can now access the nginx Service.
 
-```
-$ kubectl run --namespace=advanced-policy-demo access --rm -ti --image busybox /bin/sh
-Waiting for pod advanced-policy-demo/access-472357175-y0m47 to be running, status is Pending, pod ready: false
+### 4. Deny all egress traffic
 
-If you don't see a command prompt, try pressing enter.
+Enable egress isolation on the namespace by deploying a [default deny all egress traffic policy](https://Kubernetes.io/docs/concepts/services-networking/network-policies/#4-deny-all-egress-traffic).
 
-/ # wget -q --timeout=5 nginx -O -
-...
-/ # ping google.com
-PING google.com (216.58.219.206): 56 data bytes
-64 bytes from 216.58.219.206: seq=0 ttl=61 time=14.365 ms
+```shell
+kubectl create -f - <<EOF
+apiVersion: networking.Kubernetes.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-egress
+  namespace: advanced-policy-demo
+spec:
+  podSelector:
+    matchLabels: {}
+  policyTypes:
+  - Egress
+EOF
 ```
 
-### Prevent outgoing connections from pods
+#### Verify Access - Denied All Egress
 
-Kubernetes NetworkPolicy does not provide a way to prevent outgoing connections from pods.  However,
-Calico does. In this section we'll create egress policies using `calicoctl`
-that allow the outgoing connections in the `advanced-policy-demo` namespace we
-want; all other egress traffic will be denied.
+Now any ingress or egress traffic which is not explicitly allowed by a policy will be denied.
 
-To do this, we'll need to create a policy which selects all pods in the namespace, and allows
-egress traffic to other pods in the namespace.
+We can see that this is the case by switching over to our "access" pod in the
+namespace and attempting to lookup nginx or access google.com.
 
+```shell
+/ # nslookup nginx
+Server:    10.96.0.10
+Address 1: 10.96.0.10
+
+
+nslookup: can't resolve 'nginx'
+/ # wget -q --timeout=5 google.com -O -
+wget: bad address 'google.com'
 ```
-calicoctl apply -f - <<EOF
-apiVersion: v1
-kind: policy
+
+> **Note**: The `nslookup` command can take a minute or more to timeout.
+{: .alert .alert-info}
+
+### 5. Allow DNS egress traffic
+
+Run the following to create a `NetworkPolicy` which allows DNS egress traffic
+from any pods in the `advanced-policy-demo` namespace.
+
+```shell
+kubectl create -f - <<EOF
+apiVersion: networking.Kubernetes.io/v1
+kind: NetworkPolicy
 metadata:
-  name: advanced-policy-demo.allow-egress
+  name: allow-dns-access
+  namespace: advanced-policy-demo
 spec:
-  selector: calico/k8s_ns == 'advanced-policy-demo'
-  order: 500
+  podSelector:
+    matchLabels: {}
+  policyTypes:
+  - Egress
   egress:
-  - action: allow
-    destination:
-      selector: calico/k8s_ns == 'advanced-policy-demo'
+  - to:
```

Comment:
What do you think about maybe adding a label selector to this policy which selects the kube-system Namespace for the example?

Reply:
I discussed this with @tmjd and I think it is better to enable just DNS traffic because it is more secure than allowing all traffic to kube-system.  Erik was stating that enabling namespace selector AND ports is not allowed for egress.  If it was, then I'd be game for that.

Reply:
Yeah, I was suggesting doing an AND between namespaceSelector and ports.

e.g.

```yaml
egress:
  - to:
    - namespaceSelector:
        matchLabels: 
          name: kube-system
    ports:
    - protocol: UDP
      port: 53
    
```

Reply:
One more thing we need to do for this - the `kube-system` namespace isn't going to be labeled by default.

So, we need to ensure the user puts that label on the namespace.



Essentially, they'll need to run:

```
kubectl label namespace kube-system name=kube-system
```

Reply:
done